scipy/sparse/tests/test_base.py

```diff
@@ -4300,7 +4300,10 @@ def spmatrix(self, D, **kwargs):
                                            zero_pos[0][k],
                                            zero_pos[1][k])
 
-        arg1 = self._arg1_for_noncanonical(M)
+        if sorted_indices:
+            arg1 = self._arg1_for_noncanonical(M, sorted_indices)
+        else:
+            arg1 = self._arg1_for_noncanonical(M)
```

Comment:
`TestCOONonCanonical` also inherits from `_NonCanonicalMixin`. Without this condition, its  `_arg1_for_noncanonical` gets too many arguments. We could do the following instead:
``` python
class TestCOONonCanonical(_NonCanonicalMixin, TestCOO):
    def _arg1_for_noncanonical(self, M, sorted_indices):
        ...
```
but I did not want to change the signature of methods that are not involved (`sorted_indices` would be unused here).
Another solution would be to override `spmatrix` in `_NonCanonicalCSMixin`, which would create a lot of code duplication.